app/src/main/java/campus/tech/kakao/contacts/DetailActivity.kt

```diff
@@ -0,0 +1,107 @@
+     * - `detailEmailTextView` : 메일을 나타내는 TextView
+     * - `detailBirthdayTextView` : 생일을 나타내는 TextView
+     * - `detailGenderTextView` : 성별을 나타내는 TextView
+     * - `detailMemoTextView` : 메모를 나타내는 TextView
+     * - `detailEmailLayout` : 메일 ConstraintLayout
+     * - `detailBirthdayLayout` : 생일 ConstraintLayout
+     * - `detailGenderLayout` : 성별 ConstraintLayout
+     * - `detailMemoLayout` : 메모 ConstraintLayout
+     */
+    private fun initViews() {
+        detailNameTextView = findViewById(R.id.detail_name_text_view)
+        detailPhoneNumTextView = findViewById(R.id.detail_phone_num_text_view)
+        detailEmailTextView = findViewById(R.id.detail_email_text_view)
+        detailBirthdayTextView = findViewById(R.id.detail_birthday_text_view)
+        detailGenderTextView = findViewById(R.id.detail_gender_text_view)
+        detailMemoTextView = findViewById(R.id.detail_memo_text_view)
+
+        detailEmailLayout = findViewById(R.id.detail_email_layout)
+        detailBirthdayLayout = findViewById(R.id.detail_birthday_layout)
+        detailGenderLayout = findViewById(R.id.detail_gender_layout)
+        detailMemoLayout = findViewById(R.id.detail_memo_layout)
+    }
+
+    /**
+     * 전달된 contact 정보를 view와 layout에 설정하는 함수.
+     */
+    private fun setContactDetails() {
+        val contact: Contact? =
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                intent.getParcelableExtra("contact", Contact::class.java)
+            } else {
+                intent.getParcelableExtra("contact")
+            }
+
+        contact?.let {
+            setDetailTextViews(it)
+            setDetailLayoutVisibility(it)
+        }
+    }
+
+    /**
+     * contact 정보로 TextView의 text를 설정하는 함수.
+     *
+     * @param contact RegisterActivity로 부터 받은 Contact 객체.
+     */
+    private fun setDetailTextViews(contact: Contact) {
+        detailNameTextView.text = contact.name
+        detailPhoneNumTextView.text = contact.phoneNum
+        detailEmailTextView.text = contact.email ?: ""
+        detailBirthdayTextView.text = contact.birthday ?: ""
+        detailGenderTextView.text =
+            when (contact.gender) {
+                1 -> "여성"
+                2 -> "남성"
+                else -> ""
+            }
+        detailMemoTextView.text = contact.memo ?: ""
+    }
+
+    /**
+     * contact 정보로 layout의 visibility를 설정하는 함수.
+     *
+     * 빈 정보는 보이지 않도록 설정.
+     *
+     * @param contact RegisterActivity로 부터 받은 Contact 객체.
+     */
+    private fun setDetailLayoutVisibility(contact: Contact) {
+        detailEmailLayout.visibility = if (contact.email.isEmpty()) View.GONE else View.VISIBLE
+        detailBirthdayLayout.visibility = if (contact.birthday.isEmpty()) View.GONE else View.VISIBLE
+        detailGenderLayout.visibility = if (contact.gender == 0) View.GONE else View.VISIBLE
+        detailMemoLayout.visibility = if (contact.memo.isEmpty()) View.GONE else View.VISIBLE
```

Comment:
요부분은 View 의 extension function 중 하나인 [isVisible](https://developer.android.com/reference/kotlin/androidx/core/view/package-summary#(android.view.View).isVisible()) 을 통해 쉽게 컨트롤 가능합니다.

```suggestion
        detailEmailLayout.isVisible = !contact.email.isEmpty()
        detailBirthdayLayout.isVisible = !contact.birthday.isEmpty()
        detailGenderLayout.isVisible = contact.gender != 0
        detailMemoLayout.isVisible = !contact.memo.isEmpty()
```

Reply:
또한 Gender 를 0 과 1 로 표현하기보다 enum class 를 사용하기를 권장드려요.
```
enum class Gender {
    MALE, FEMALE
}
```